### PR TITLE
OnnxExporter

### PR DESCRIPTION
**Pull request by skanjila**
# Code Pull Requests

Please provide the following:

- a simple set of classes to do onnx export of pytorch models

# Documentation Pull Requests

Note that the documentation HTML files are in `docs/` while the Markdown sources are in `mkdocs/docs`.

If you are proposing a modification to the documentation you should change only the Markdown files.

`api.md` is automatically generated from the docstrings in the code, so if you want to change something in that file, first modify `ludwig/api.py` docstring, then run `mkdocs/code_docs_autogen.py`, which will create `mkdocs/docs/api.md` .

### CODE REVIEW

**Review by saad-palapa — APPROVED** [review]

**Review by saad-aimg — COMMENTED** [review]
Great job!

**Comment by justinxzhao** [comment]
Hi @skanjila, it looks like there's a couple of pre-commit issues to resolve: https://results.pre-commit.ci/run/github/163346054/1699499452.zgxnE5vuTJujO_3rx1xLkQ

File: ludwig/model_export/base_model_exporter.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+class LudwigTorchWrapper(torch.nn.Module):
+    def __init__(self, model):
+        super().__init__()
+        self.model = model
+
+    def forward(self, x):
+        return self.model({"image_path": x})
+
+
+class BaseModelExporter(ABC):
```
Comment (justinxzhao) [inline ludwig/model_export/base_model_exporter.py:17]: Could we add docstrings to these new classes?

File: ludwig/model_export/onnx_exporter.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+from abc import ABC, abstractmethod
+
+import onnx
+import torch
+
+from ludwig.api import LudwigModel
+from ludwig.model_export.base_model_exporter import LudwigTorchWrapper
+
+
+class OnnxExporter(ABC):
```
Comment (justinxzhao) [inline ludwig/model_export/onnx_exporter.py:11]: Did you mean to inherit from `BaseModelExporter`?

File: ludwig/model_export/base_model_exporter.py (hunk 1 above)
Comment (skanjila) [reply ludwig/model_export/base_model_exporter.py:17]: Added

File: ludwig/model_export/onnx_exporter.py (hunk 2 above)
Comment (skanjila) [reply ludwig/model_export/onnx_exporter.py:11]: yes I did, just forgot to put that in so added this just now

**Pull request merged**